polybar-scripts/updates-fedora/updates-fedora.sh

```diff
@@ -1,14 +1,33 @@
 #!/bin/sh
 
-dnf=$(env LC_ALL=C sudo dnf upgrade --refresh --assumeno 2> /dev/null)
+# Choose the dnf command this script use to retrieve updates information
+#
+# 1 - updateinfo 
+#     Shows uninstalled updates but doesn't show new dependencies
+# 2 - update (need sudo privilege)
+#     Shows uninstalled updates and new dependencies
 
-upgrade=$(echo "$dnf" | grep '^Upgrade ' | awk '{ print $2 }')
-install=$(echo "$dnf" | grep '^Install ' | awk '{ print $2 }')
+mode="$1"
 
-updates=$(( upgrade + install ))
+case "$mode" in
+  1)
+    updates=$(dnf -q updateinfo --list --updates 2> /dev/null | wc -l )
+    ;;
+  2)
+    dnf=$(env LC_ALL=C sudo dnf upgrade --assumeno 2> /dev/null)
+    
+    upgrade=$(echo "$dnf" | grep '^Upgrade ' | awk '{ print $2 }')
+    install=$(echo "$dnf" | grep '^Install ' | awk '{ print $2 }')
+
+    updates=$(( upgrade + install ))
+    ;;
+  *)
+    updates=$(dnf -q updateinfo --list --updates 2> /dev/null | wc -l )
```

Comment:
I tested it on Fedora. `dnf -q updateinfo --list --updates 2> /dev/null` looks good. I am still not sure what our driver behind `dnf upgrade --assumeno` was. We should go with your solution, remove everything else to keep it clean and simple.